src/sink/kinesis.rs

```diff
@@ -162,7 +167,7 @@ impl Kinesis {
 
         while !buffer.is_empty() {
             let put_records_input = PutRecordsInput {
-                records: self.buffer.clone(),
+                records: buffer.clone(),
```

Comment:
I wonder if you couldn't get higher throughput on this sink by relying only on `self.buffer`. Each time we go to `publish_buffer` we attempt to ship the whole thing, filter those records that were successfully sent etc etc. It would mean, I guess, potentially higher latency for any given buffer record, since you may not be able to keep the loop on 168. 

Might not be possible depending on the ownership semantics of PutRecordsInput?

Reply:
As I recall it wasn't possible due to PutRecordsInput.

Reply:
https://rusoto.github.io/rusoto/rusoto_kinesis/struct.PutRecordInput.html

It is just a Vec<u8>.  Though the structure is fully public.  I could use it as my buffer.